lib/ecto/schema.ex

```diff
@@ -525,9 +527,13 @@ defmodule Ecto.Schema do
     :type,
     :where,
     :references,
-    :skip_default_validation
+    :skip_default_validation,
+    :mode
   ]
 
+  @default_mode :readwrite
+  @allowed_modes [:readwrite, :readonly]
```

Comment:
We use `read_only` in the repo, so at least here we should be consistent and have `read_write` and `read_only`?